zap/src/main/java/org/zaproxy/zap/extension/ascan/ActiveScanPanel.java

```diff
@@ -229,19 +255,39 @@ public void clearFinishedScans() {
     }
 
     @Override
-    protected JScrollPane getWorkPanel() {
-        if (jScrollPane == null) {
-            jScrollPane = new JScrollPane();
-            jScrollPane.setViewportView(getMessagesTable());
+    protected JPanel getWorkPanel() {
+        if (mainPanel == null) {
+            mainPanel = new JPanel(new BorderLayout());
+            tabbedPane = new JTabbedPane();
+            if (jScrollPane == null) {
+                jScrollPane = new JScrollPane();
+                jScrollPane.setName("ActiveScanMessagePane");
+                jScrollPane.setViewportView(getMessagesTable());
+            }
+            if (filterPane == null) {
+                filterPane = new JScrollPane();
+                filterPane.setName("FilterMessagePane");
+                filterPane.setViewportView(getFilterMessageTable());
+            }
+            tabbedPane.add(
+                    Constant.messages.getString("ascan.panel.tab.scannedMessages"), jScrollPane);
+            tabbedPane.add(
+                    Constant.messages.getString("ascan.panel.tab.filteredMessages"), filterPane);
+            tabbedPane.setSelectedIndex(0);
+            mainPanel.add(tabbedPane);
         }
-        return jScrollPane;
+        return mainPanel;
     }
 
     private void resetMessagesTable() {
         getMessagesTable().setModel(EMPTY_RESULTS_MODEL);
     }
 
-    private HistoryReferencesTable getMessagesTable() {
+    private void resetFilterMessageTable() {
+        getFilterMessageTable().setModel(EMPTY_FILTER_MESSAGE_MODEL);
+    }
+
+    private ZapTable getMessagesTable() {
```

Comment:
This can still return `HistoryReferencesTable`.

Reply:
hmm correct.